prisma/migrations/20250402214855_enable_rls/migration.sql

```diff
@@ -0,0 +1,274 @@
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+-- Challenges
+CREATE POLICY "Users can view their own challenges" ON "Challenges"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own challenges" ON "Challenges"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+CREATE POLICY "Users can update their own challenges" ON "Challenges"
+  FOR UPDATE USING (auth.uid() = "userId");
+CREATE POLICY "Users can delete their own challenges" ON "Challenges"
+  FOR DELETE USING (auth.uid() = "userId");
+
+-- DevicesAndLocations
+CREATE POLICY "Users can view their own devices and locations" ON "DevicesAndLocations"
+  FOR SELECT USING (auth.uid() = "userId" OR "userId" IS NULL);
+CREATE POLICY "Users can insert devices and locations" ON "DevicesAndLocations"
+  FOR INSERT WITH CHECK (auth.uid() = "userId" OR "userId" IS NULL);
+
+-- Sessions
+CREATE POLICY "Users can view their own sessions" ON "Sessions"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can update their own sessions" ON "Sessions"
+  FOR UPDATE USING (auth.uid() = "userId");
+CREATE POLICY "Users can delete their own sessions" ON "Sessions"
+  FOR DELETE USING (auth.uid() = "userId");
+
+-- Membership-related policies
+-- Organizations
+CREATE POLICY "Users can view organizations they are members of" ON "Organizations"
+  FOR SELECT USING (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."organizationId" = "Organizations"."id"
+      AND "Memberships"."userId" = auth.uid()
+    )
+  );
+
+CREATE POLICY "Users with owner role can update their organizations" ON "Organizations"
+  FOR UPDATE USING (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."organizationId" = "Organizations"."id"
+      AND "Memberships"."userId" = auth.uid()
+      AND "Memberships"."role" = 'OWNER'
+    )
+  );
+
+-- Teams
+CREATE POLICY "Users can view teams they are members of" ON "Teams"
+  FOR SELECT USING (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."teamId" = "Teams"."id"
+      AND "Memberships"."userId" = auth.uid()
+    )
+  );
+
+CREATE POLICY "Users with owner/admin role can update their teams" ON "Teams"
+  FOR UPDATE USING (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."teamId" = "Teams"."id"
+      AND "Memberships"."userId" = auth.uid()
+      AND "Memberships"."role" IN ('OWNER', 'ADMIN')
+    )
+  );
+
+-- Applications
+CREATE POLICY "Team members can view applications" ON "Applications"
+  FOR SELECT USING (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."userId" = auth.uid()
+      AND "Memberships"."teamId" = "Applications"."teamId"
+    )
+  );
+
+CREATE POLICY "Team owners/admins can insert applications" ON "Applications"
+  FOR INSERT WITH CHECK (
+    EXISTS (
+      SELECT 1 FROM "Memberships"
+      WHERE "Memberships"."userId" = auth.uid()
+      AND "Memberships"."teamId" = NEW."teamId"
```

Comment:
I am getting error: `missing FROM-clause entry for table "new"`
```suggestion
      AND "Memberships"."teamId" = "Applications"."teamId"
```